### PR TITLE
Add test for async client

### PR DESCRIPTION
**Pull request by joelwurtz**
In relation to https://github.com/php-http/httplug/pull/77

### CODE REVIEW

File: src/HttpAdapterTest.php [hunk 1]
```diff
@@ -130,6 +130,45 @@ public function testSendRequest($method, $uri, array $headers, $body)
...
+     * @dataProvider requestProvider
+     * @group        integration
+     */
+    public function testAsyncSendRequest($method, $uri, array $headers, $body)
+    {
+        if ($body != null) {
+            $headers['Content-Length'] = (string)strlen($body);
+        }
+
+        if (!($this->httpAdapter instanceof HttpAsyncClient)) {
```
Comment (sagikazarmark) [inline src/HttpAdapterTest.php]: Shouldn't this be the first thing?

Comment (joelwurtz) [inline src/HttpAdapterTest.php]: Hum yes :)

**Comment by sagikazarmark** [comment]
Shouldn't we create two separate tests instead?

Also, shouldn't we refactor tests to reflect the change from adapter to client?

**Comment by joelwurtz** [comment]
Hum yes may be better, i just go for the fastest right now to validate guzzle implementation, but will do this tests in a separate class

**Comment by joelwurtz** [comment]
This should be good.

This need the merge of https://github.com/php-http/discovery/pull/15

**Comment by joelwurtz** [comment]
Updated with new repository for async, should be good to merge.

**Comment by sagikazarmark** [comment]
Can you please squash commits? I think it makes sense in this case.

**Comment by joelwurtz** [comment]
done

**Comment by sagikazarmark** [comment]
Thanks

**Pull request merged**